File: packages/js/examples/resolve.ts
// Read-only: resolve a wallet address from a name.
// Run: npx tsx examples/resolve.ts john
import { SAFROCHAIN_TESTNET, SafHandleClient } from "@safrochaindev/safhandle";

async function main() {
  const input = process.argv[2] ?? "john";

  // Testnet's contract address is baked into the SDK. Set SAFHANDLE_CONTRACT to
  // point at a different deployment via the `custom` network.
  const contract = process.env.SAFHANDLE_CONTRACT;
  const client = await SafHandleClient.connect(
    contract
      ? {
          network: "custom",
          rpcEndpoint: SAFROCHAIN_TESTNET.rpcEndpoint,
          contractAddress: contract,
        }
      : { network: "testnet", rpcEndpoint: SAFROCHAIN_TESTNET.rpcEndpoint },
  );

  const address = await client.lookup(input);
  if (address === null) {
    console.log(`"${input}" is not registered.`);
    return;
  }

  const record = await client.getAddress(input);
  console.log(`${input} → ${address}`);
  console.log(`  type: ${record.record_type}, key: ${record.normalized_key}`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: packages/dart/example/resolve.dart
// Read-only: resolve a wallet address from a name.
//
// Run: dart run example/resolve.dart john
// Point at a different deployment with SAFHANDLE_CONTRACT.
import 'dart:io';

import 'package:safhandle/safhandle.dart';

Future<void> main(List<String> args) async {
  final input = args.isNotEmpty ? args.first : 'john';

  // Testnet's contract address is baked into the SDK. Set SAFHANDLE_CONTRACT to
  // point at a different deployment via the `custom` network.
  final contract = Platform.environment['SAFHANDLE_CONTRACT'];
  final client = SafHandleClient.connect(
    contract != null
        ? ConnectOptions.custom(
            restEndpoint: safrochainTestnet.restEndpoint,
            contractAddress: contract,
          )
        : ConnectOptions.testnet(),
  );

  try {
    final address = await client.lookup(input);
    if (address == null) {
      stdout.writeln('"$input" is not registered.');
      return;
    }

    final record = await client.getAddress(input);
    stdout.writeln('$input → $address');
    stdout
        .writeln('  type: ${record.recordType}, key: ${record.normalizedKey}');
  } finally {
    client.close();
  }
}
